src/server/bg_services/log_replication_scanner.js

```diff
@@ -120,6 +120,13 @@ class LogReplicationScanner {
         }));
     }
 
+    /**
+     * process_candidates takes the source and destination buckets and the candidates
+     * and process them depending upon the kind of candidate.
+     * @param {*} src_bucket source bucket
+     * @param {*} dst_bucket destination bucket
+     * @param {Array<{ key: string, action: 'copy' | 'delete', time: Date }>} candidates 
```

Comment:
this row repeats 4 times, can you define this type?

Reply:
Makes sense. Should I add it to `nb.d.ts`?

Reply:
hmmm I think you can

Reply:
Added types ion `nb.d.ts`.